### PR TITLE
Emit note during cmake if aarch64 host supports crc instructions, but option not selected.

### PR DESCRIPTION
**Pull request by swalk-cavium**
The ability to use the ARMv8 crc instructions was added by PR#7556.  They
are enabled only if an option is added to the cmake command line. 

This makes a minor change to the HPHPCompiler.cmake file to emit a note
if the build host supports the crc instruction and it has not been selected.

### CODE REVIEW

File: hphp/tools/gdb/gdbutils.py [hunk 1]
```diff
@@ -300,3 +304,11 @@ def deref(val):
         return val.cast(rawtype(val.type))
     else:
         return deref(p.referenced_value())
+
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: +1 line whitespace here to match the rest of the file.

File: hphp/tools/gdb/gdbutils.py [hunk 2]
```diff
@@ -300,3 +304,11 @@ def deref(val):
...
+def get_arch(self):
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: This function needs to be documented with a triple-quote docstring, as all the functions above.

File: hphp/tools/gdb/gdbutils.py [hunk 3]
```diff
@@ -300,3 +304,11 @@ def deref(val):
...
+    try:
+        arch = gdb.selected_frame().architecture().name()
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: Just return it here...

File: hphp/tools/gdb/gdbutils.py [hunk 4]
```diff
@@ -300,3 +304,11 @@ def deref(val):
...
+    except:
+        arch = re.search('\(currently (.*)\)', gdb.execute('show architecture', to_string=True)).group(1)
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: ...and here...

File: hphp/tools/gdb/gdbutils.py [hunk 5]
```diff
@@ -300,3 +304,11 @@ def deref(val):
...
+    return arch
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: And you can skip having an `arch` variable.

File: hphp/tools/gdb/gdbutils.py (hunk 4 above)
Comment (mxw) [reply hphp/tools/gdb/gdbutils.py]: Also, wrap to 80 chars.  Maybe

```python
arch = re.search(
    '\(currently (.*)\)',
    gdb.execute('show architecture', to_string=True)
).group(1)
```

File: hphp/tools/gdb/gdbutils.py [hunk 6]
```diff
@@ -126,7 +126,11 @@ def string_data_val(val, keep_case=True):
     else:
         data = val['m_data']
 
-    s = data.string('utf-8', 'ignore', val['m_len'])
+    try:
+        s = data.string('utf-8', 'ignore', val['m_len'])
+    except:
+        s = "Unknown"
+
```
Comment (mxw) [inline hphp/tools/gdb/gdbutils.py]: Undo this change.

Comment (swalk-cavium) [reply hphp/tools/gdb/gdbutils.py]: @mxw - Actually, without this change the lookup fails hard and stops crawling the stack.
Example below:

#18 0xffffffffd4b0 @ 0x05c2c91c: HPHP::ObjectData::release() at /mnt/cdn/hhvm/play.083017a/hhvm/hphp/runtime/base/object-data.cpp:203
#19 0xffffacefff90 @ 0x14000228: [PHP] foo() at /mnt/cdn/hhvm/play.083017a/hhvm/hphp/test/quick/setg-dtor.php
#20 0xffffacefffc0 @ 0x0f000918: [PHP] <pseudomain>() at /mnt/cdn/hhvm/play.083017a/hhvm/hphp/test/quick/setg-dtor.php

Traceback (most recent call last):
  File "/mnt/cdn/hhvm/play.083017a/hhvm/hphp/tools/gdb/gdbutils.py", line 55, in wrapped
    return func(*args, **kwds)
  File "/mnt/cdn/hhvm/play.083017a/hhvm/hphp/tools/gdb/stack.py", line 115, in invoke
    idx=i, ar=fp.cast(ar_type), rip=rip)))
  File "/mnt/cdn/hhvm/play.083017a/hhvm/hphp/tools/gdb/frame.py", line 163, in create_php
    func_name = nameof(func)
  File "/mnt/cdn/hhvm/play.083017a/hhvm/hphp/tools/gdb/nameof.py", line 34, in nameof
    return string_data_val(deref(sd))
  File "/mnt/cdn/hhvm/play.083017a/hhvm/hphp/tools/gdb/gdbutils.py", line 130, in string_data_val
    s = data.string('utf-8', 'ignore', val['m_len'])
OverflowError: signed integer is greater than maximum

Python Exception <class 'OverflowError'> signed integer is greater than maximum: 
Error occurred in Python command: signed integer is greater than maximum

**Comment by swalk-cavium** [comment]
Didn't create branch properly.  Intertwingled two separate commits.  These comments will
be applied to PR#7985.  The cmake file changes will be submitted separately.
